pkg/shell/topnav.jsx

```diff
@@ -168,6 +171,8 @@ export class TopNav extends React.Component {
             </DropdownItem>,
         );
 
+        const D = this.context;
```

Comment:
Can this be moved up in the file, so that a reverse search for the identifier finds the declaration? Also see above for giving this a better name, or perhaps we just use `this.context` as-is?

Reply:
Oh, I didn't realize it is declared after first use! Thanks for catching this!

Reply:
Fixed by rewriting to function components.